celery/worker/loops.py

```diff
@@ -28,11 +28,23 @@ def _quick_drain(connection, timeout=0.1):
 
 
 def _enable_amqheartbeats(timer, connection, rate=2.0):
-    if connection:
-        tick = connection.heartbeat_check
-        heartbeat = connection.get_heartbeat_interval()  # negotiated
-        if heartbeat and connection.supports_heartbeats:
-            timer.call_repeatedly(heartbeat / rate, tick, (rate,))
+    heartbeat_error = [None]
```

Comment:
I'm not seeing why this is a `list` instead of just `None`. What benefit is there by making this a list and sometimes modifying the first parameter?

Reply:
It preallocates the list with one item.
It also avoids confusion and possible bugs with `list.append()`.

It's possible that this should be a list since it is reference counted and can be mutated from anywhere.

I'm OK with this, but I'd like the OP to verify.

Reply:
Sure, although the pre-allocation seems like an unnecessary optimization.

My question is really -- why isn't this just a bare value?

Reply:
@jagguli Care to explain?

Reply:
don't you have to pre-allocate a mutable construct to modify it in `tick`? I'll wait for @jagguli 's explanation but I think that is it.

Reply:
pre-allocating because somewhere else `if not heartbeat:` will be true, there is more to it I tried a couple of different vairations and this is the simplest

Reply:
Jaggull, can you update tests for this PR?